lms/djangoapps/edraak_forus/views.py

```diff
@@ -0,0 +1,248 @@
+import logging
+import json
+from urllib import urlencode
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.contrib.auth import logout, login
+from django.contrib.auth.models import User, AnonymousUser
+from django.utils.translation import check_for_language, ugettext as _
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from util.json_request import JsonResponse
+
+from student_account.views import _get_form_descriptions
+from django.views.decorators.csrf import csrf_exempt
+from django.utils.decorators import method_decorator
+from django.core.exceptions import ValidationError
+from student.cookies import delete_logged_in_cookies
+
+from edxmako.shortcuts import render_to_response
+from third_party_auth import pipeline
+from django.views.generic import View
+
+from openedx.core.djangoapps.user_api.preferences.api import set_user_preference
+from lang_pref import LANGUAGE_KEY
+
+from xmodule.modulestore.django import modulestore
+
+from django_locale.trans_real import LANGUAGE_SESSION_KEY
+
+from student.models import AlreadyEnrolledError
+from student.helpers import enroll, get_next_url_for_login_page, InvalidCourseIdError
+
+from openedx.core.djangoapps.user_api.views import RegistrationView
+
+from course_modes.helpers import get_mktg_enroll_success_redirect
+
+from .models import ForusProfile
+from .helpers import validate_forus_params, forus_error_redirect
+
+log = logging.getLogger(__name__)
+
+
+class UpdateLanguageView(View):
+    def get(self, request):
+        self.update_language()
+        return redirect(reverse('forus_v1_auth_validate') + '?' + request.META['QUERY_STRING'])
+
+    def update_language(self):
+        url_lang = self.request.GET.get('lang')
+        if check_for_language(url_lang):
+            self.request.session[LANGUAGE_SESSION_KEY] = url_lang
```

Comment:
😁

On Tue, Feb 2, 2016 at 10:58 AM, Ali <notifications@github.com> wrote:

> In lms/djangoapps/edraak_forus/views.py
> <https://github.com/Edraak/edx-platform/pull/144#discussion_r51540064>:
>
> > +
> > +from .models import ForusProfile
> > +from .helpers import validate_forus_params, forus_error_redirect
> > +
> > +log = logging.getLogger(__name__)
> > +
> > +
> > +class UpdateLanguageView(View):
> > +    def get(self, request):
> > +        self.update_language()
> > +        return redirect(reverse('forus_v1_auth_validate') + '?' + request.META['QUERY_STRING'])
> > +
> > +    def update_language(self):
> > +        url_lang = self.request.GET.get('lang')
> > +        if check_for_language(url_lang):
> > +            self.request.session[LANGUAGE_SESSION_KEY] = url_lang
>
> Or you know what, change language should be done in a middleware.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Edraak/edx-platform/pull/144/files#r51540064>.
>



-- 
Omar Al-Ithawi

Senior Engineer, Edraak.org <http://www.edraak.org/>


Queen Rania Foundation

for Education & Development

T +962 6 4016464  Ext. 700

M +962 7 90574405

E oithawi@qrf.org

<https://twitter.com/edraak> <https://www.facebook.com/edraakorg>
<http://www.youtube.com/user/edraakorg> <http://instagram.com/edraakorg>